### PR TITLE
fix: Click on following/followers tab with a disconnected account

### LINKED ISSUE #10259: Click on following/followers tab with a disconnected account?
what should happen if i click on it with a disconnected account? 

[Gravação de tela de 07-05-2024 09:39:59.webm](https://github.com/kodadot/nft-gallery/assets/36627808/eb627f1f-454e-4078-9a89-e532a6d7bd92)

_Originally posted by @prury in https://github.com/kodadot/nft-gallery/issues/10250#issuecomment-2098316324_

### PR DESCRIPTION
**Pull request by Jarsen136**
**Thank you for your contribution** to the [Koda - Generative Art Marketplace](https://kodadot.xyz).

👇 __ Let's make a quick check before the contribution.

## PR Type

- [x] Bugfix

## Needs Design check

- @exezbcz please review

## Needs QA check

- @kodadot/qa-guild please review

## Context

- [x] Closes #10259

#### Did your issue had any of the "$" label on it?

- [x] My DOT address: [Payout](https://canary.kodadot.xyz/dot/transfer/?target=16SjUbGKSdjCdWTy3NNT3JxbRVGGqD4mwkHpc6BD9U2Rp29Z)

## Screenshot 📸

- [ ] My fix has changed **something** on UI; a screenshot is best to understand changes for others.

### CODE REVIEW

**Comment by prury** [comment]
same happens here:

image

**Comment by Jarsen136** [comment]
> same happens here:

The problem is that the testing address follows the address on both base and ahp. For now the support for `evm` related would be implemented only on the `feat/evm/v0` branch. 

In short, it should be fixed on this issue https://github.com/kodadot/nft-gallery/issues/10466

Could you test with another address?

image

**Comment by prury** [comment]
> > same happens here:
> 
> The problem is that the testing address follows the address on both base and ahp. For now the support for `evm` related would be implemented only on the `feat/evm/v0` branch.
> 
> In short, it should be fixed on this issue #10466
> 
> Could you test with another address? @prury
> image

oh ok, it works then

**Comment by Jarsen136** [comment]
@vikiival @hassnian could you code review?

File: services/profile.ts [hunk 1]
```diff
@@ -77,7 +77,11 @@ const invalidSignatureErrorHandler = (error: FetchError) => {
 }
 
 export const toSubstrateAddress = (address: string) =>
-  isEthereumAddress(address) ? address : formatAddress(address, 42)
+  !address
+    ? ''
+    : isEthereumAddress(address)
+      ? address
+      : formatAddress(address, 42)
```
Comment (hassnian) [inline services/profile.ts:84]:
```suggestion
        if(!address) {
             return ''
         }
      
      return isEthereumAddress(address)
      ? address
      : formatAddress(address, 42)
```

File: components/profile/FollowButton.vue [hunk 2]
```diff
@@ -65,12 +71,13 @@ const unfollowConfig = computed<ButtonConfig>(() => ({
       loading.value = false
       return
     })
-    await unfollow({
-      initiatorAddress: accountId.value,
-      targetAddress: props.target,
-      signature: signaturePair.signature,
-      message: signaturePair.message,
-    })
+    signaturePair &&
```
Comment (hassnian) [inline components/profile/FollowButton.vue:74]:
```suggestion
    if(!signaturePair) {
    loading.value = false
    return 
    }
```

File: components/profile/FollowButton.vue [hunk 3]
```diff
@@ -32,26 +34,30 @@ const { data: isFollowingThisAccount, refresh: refreshFollowingStatus } =
...
-      return
+    doAfterLogin({
+      onLoginSuccess: async () => {
+        loading.value = true
+        const signaturePair = await getSignaturePair().catch((e) => {
+          toast(e.message)
+          loading.value = false
+          return
+        })
+        signaturePair &&
```
Comment (hassnian) [inline components/profile/FollowButton.vue:46]:
```suggestion
    if(!signaturePair) {
    loading.value = false
    return 
    }
```

File: components/profile/follow/UserRow.vue [hunk 4]
```diff
@@ -92,27 +94,29 @@ const followConfig: ButtonConfig = {
...
-        initiatorAddress: accountId.value,
-        targetAddress: props.user.address,
-        signature: signaturePair.signature,
-        message: signaturePair.message,
-      }).catch(() => {
-        openProfileCreateModal()
-      }))
-    showFollowing.value = true
-    refresh()
+        signaturePair &&
```
Comment (hassnian) [inline components/profile/follow/UserRow.vue:104]:
```suggestion
    if(!signaturePair) {
    loading.value = false
    return 
    }
```

File: services/profile.ts [hunk 5]
```diff
@@ -77,7 +77,11 @@ const invalidSignatureErrorHandler = (error: FetchError) => {
 }
 
 export const toSubstrateAddress = (address: string) =>
-  isEthereumAddress(address) ? address : formatAddress(address, 42)
+  !address
```
Comment (hassnian) [inline services/profile.ts:80]: we should change the name of this function, is not really `toSubstrateAddress` anymore , now that we've added evm

File: components/profile/follow/UserRow.vue [hunk 6]
```diff
@@ -55,6 +55,8 @@ import { getss58AddressByPrefix } from '@/utils/account'
 import { openProfileCreateModal } from '@/components/profile/create/openProfileModal'
 const { accountId } = useAuth()
 const { $i18n } = useNuxtApp()
+const instance = getCurrentInstance()
+const { doAfterLogin } = useDoAfterlogin(instance)
```
Comment (hassnian) [inline components/profile/follow/UserRow.vue:59]:
```suggestion
const { doAfterLogin } = useDoAfterlogin(getCurrentInstance())
```

File: components/profile/FollowButton.vue [hunk 7]
```diff
@@ -18,6 +18,8 @@ const { accountId } = useAuth()
 const { getSignaturePair } = useVerifyAccount()
 const isHovered = useElementHover(buttonRef)
 const { toast } = useToast()
+const instance = getCurrentInstance()
+const { doAfterLogin } = useDoAfterlogin(instance)
```
Comment (hassnian) [inline components/profile/FollowButton.vue:22]:
```suggestion
const { doAfterLogin } = useDoAfterlogin(getCurrentInstance())
```

**Review by hassnian — CHANGES_REQUESTED** [review]
api calls are not called if `signaturePair` is nullish , yet the function continues as if it was successful

**Comment by Jarsen136** [comment]
> if `signaturePair` is nullish the functions continue as if it was successful
> 
>  CleanShot.2024-06-22.at.08.58.38.mp4

Thanks, all updated

File: services/profile.ts (hunk 5 above)
Comment (Jarsen136) [reply services/profile.ts]: I would leave it the same as the naming change could cause code conflict between branchs.

**Comment by vikiival** [comment]
Can you change branch pls?

**Pull request closed without merging**